Zilon.Core/Zilon.Core/PersonModules/CombatActModule.cs

```diff
@@ -7,13 +20,206 @@ namespace Zilon.Core.PersonModules
+            {
+                throw new ArgumentNullException(nameof(equipments));
+            }
+
+            var actList = new List<ITacticalAct>();
+
+            var defaultAct = CreateTacticalAct(defaultActScheme, equipment: null, effects: effects, perks: perks);
+            actList.Insert(0, defaultAct);
+
+            if (equipments != null)
+            {
+                foreach (var equipment in equipments)
+                {
+                    if (equipment == null)
+                    {
+                        continue;
+                    }
+
+                    foreach (var actScheme in equipment.Acts)
+                    {
+                        var act = CreateTacticalAct(actScheme, equipment, effects, perks);
+
+                        actList.Insert(0, act);
+                    }
+                }
+            }
+
+            return actList.ToArray();
+        }
+
+        private IEnumerable<IPerk> GetPerksSafe()
+        {
+            var perks = _evolutionModule?.GetArchievedPerks();
+            if (perks == null)
+            {
+                perks = Array.Empty<IPerk>();
+            }
+
+            return perks;
+        }
+
+        private static ITacticalAct CreateTacticalAct([NotNull] ITacticalActScheme scheme,
+            [NotNull] Equipment equipment,
+            [NotNull] IEffectsModule effects,
+            [NotNull, ItemNotNull] IEnumerable<IPerk> perks)
+        {
+            var toHitModifierValue = 0;
+            var efficientModifierValue = 0;
+            var efficientRollUnmodified = scheme.Stats.Efficient;
+            CalcSurvivalHazardOnTacticalAct(effects, ref toHitModifierValue, ref efficientModifierValue);
+            CalcPerkBonusesOnTacticalAct(perks, equipment, ref toHitModifierValue, ref efficientModifierValue);
+
+            var toHitRoll = CreateTacticalActRoll(6, 1, toHitModifierValue);
+            var efficientRoll = CreateTacticalActRoll(efficientRollUnmodified.Dice,
+                efficientRollUnmodified.Count,
+                efficientModifierValue);
+
+            return new TacticalAct(scheme, efficientRoll, toHitRoll, equipment);
+        }
+
+        private static void CalcPerkBonusesOnTacticalAct([NotNull, ItemNotNull] IEnumerable<IPerk> archievedPerks,
+            [NotNull] Equipment equipment,
+            ref int toHitModifierValue,
+            ref int efficientModifierValue)
+        {
+            foreach (var perk in archievedPerks)
+            {
+                var currentLevel = perk.CurrentLevel;
+                var currentLevelScheme = perk.Scheme.Levels[currentLevel.Primary];
+
+                if (currentLevelScheme.Rules == null)
+                {
+                    continue;
+                }
+
+                for (var i = 0; i <= currentLevel.Sub; i++)
+                {
+                    foreach (var rule in currentLevelScheme.Rules)
+                    {
+                        switch (rule.Type)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5480188)